src/main/java/org/plp/implementations/plpisa/PlpFile.java

```diff
@@ -0,0 +1,132 @@
+package org.plp.implementations.plpisa;
+
+import org.plp.isa.AsmFile;
+import org.plp.isa.AsmProgram;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This is an in memory representation of an asm file {@link AsmFile}.
+ * This will map to a file path and will be part of {@link PlpProgram}
+ */
+public class PlpFile implements AsmFile {
+    private final String fileName;
+    private List<String> fileContent;
+    private final AsmProgram program;
```

Comment:
Does it seem like we don't need to store the AsmProgram pointer if we compute the path in our constructor?

Reply:
Yes going to make AsmFile independent of the AsmProgram. Program will be managing files but files itself do not need to know about the program.